meilisearch/task.py

```diff
@@ -0,0 +1,116 @@
+from time import sleep
+from datetime import datetime
+from typing import Any, Dict, List, Optional
+
+from meilisearch._httprequests import HttpRequests
+from meilisearch.config import Config
+from meilisearch.errors import MeiliSearchTimeoutError
+
+class Task:
```

Comment:
What you have done with creating a `Task` class works so this is only a suggestion for simplification. If you like the class approach better disregard this comment as both approaches work.

Because you are always using the methods in Tasks as functions and never using the class to store data for later use you can remove the class all together. This way you can call the functions directly rather than needing to create an object first. `tasks.py` would become (doc strings stripped for space):

```py
def get_tasks(config: Config, index_id: Optional[str] = None) -> Dict[str, List[Dict[str, Any]]]:
    http = HttpRequests(config)
    if index_id is None:
        return http.get(
            f'{config.paths.task}'
        )
    return http.get(
        f'{config.paths.index}/{index_id}/{config.paths.task}'
    )


def get_task(config: Config, uid: int, index_id: Optional[str] = None) -> Dict[str, Any]:
    http = HttpRequests(config)
    if index_id is None:
        return http.get(
            f'{config.paths.task}/{uid}'
        )
    return http.get(
        f'{config.paths.index}/{index_id}/{config.paths.task}/{uid}'
    )


def wait_for_task(
    config: Config,
    uid: int,
    timeout_in_ms: int = 5000,
    interval_in_ms: int = 50,
) -> Dict[str, Any]:
    start_time = datetime.now()
    elapsed_time = 0.
    while elapsed_time < timeout_in_ms:
        task = get_task(config, uid)
        if task['status'] not in ('enqueued', 'processing'):
            return task
        sleep(interval_in_ms / 1000)
        time_delta = datetime.now() - start_time
        elapsed_time = time_delta.seconds * 1000 + time_delta.microseconds / 1000
    raise MeiliSearchTimeoutError(f'timeout of ${timeout_in_ms}ms has exceeded on process ${uid} when waiting for task to be resolve.')
```

Then to call it later, with `wait_for_task` as an example, would change from:

```py
Task(self.config).wait_for_task(uid, timeout_in_ms, interval_in_ms)
```

to

```py
wait_for_task(self.config, uid, timeout_in_ms, interval_in_ms)
```

Reply:
You right, I think there is no need for a class I remove it.